tests/std/tests/P1502R1_standard_library_header_units/test.cpp

```diff
@@ -0,0 +1,1022 @@
+import <version>;
+// clang-format on
+
+#include <assert.h>
+#include <stdio.h>
+#include <force_include.hpp>
+using namespace std;
+
+int main() {
+    {
+        puts("Testing <algorithm>.");
+        constexpr int arr[]{11, 0, 22, 0, 33, 0, 44, 0, 55};
+        assert(count(begin(arr), end(arr), 0) == 4);
+        assert(ranges::count(arr, 0) == 4);
+        static_assert(count(begin(arr), end(arr), 0) == 4);
+        static_assert(ranges::count(arr, 0) == 4);
+    }
+
+    {
+        puts("Testing <any>.");
+        any a1{1729};
+        any a2{7.5};
+        a1.swap(a2);
+        assert(any_cast<double>(a1) == 7.5);
+        assert(any_cast<int>(a2) == 1729);
+    }
+
+    {
+        puts("Testing <array>.");
+#if 0 // TRANSITION, VSO-1088552 (deduction guides)
+        constexpr array arr{10, 20, 30, 40, 50};
+#else // ^^^ no workaround / workaround vvv
+        constexpr array<int, 5> arr{10, 20, 30, 40, 50};
+#endif // ^^^ workaround ^^^
+        assert(arr[2] == 30);
+        static_assert(arr[2] == 30);
+    }
+
+    {
+        puts("Testing <atomic>.");
+        atomic<int> atom{1729};
+        assert(atom.load() == 1729);
+        ++atom;
+        assert(atom.load() == 1730);
+        static_assert(atomic<int>::is_always_lock_free); // implementation-defined
+    }
+
+    {
+        puts("Testing <barrier>.");
+        static_assert(barrier<>::max() >= 5);
+        barrier b{2};
+        atomic<int> atom{0};
+        thread t1{[&] {
+            for (int i = 0; i < 5; ++i) {
+                auto token = b.arrive();
+                b.wait(move(token));
+                atom.fetch_add(1, memory_order_relaxed);
+            }
+        }};
+        thread t2{[&] {
+            for (int i = 0; i < 3; ++i) {
+                b.arrive_and_wait();
+                atom.fetch_add(1, memory_order_relaxed);
+            }
+            b.arrive_and_drop();
+        }};
+        t1.join();
+        t2.join();
+        assert(atom.load(memory_order_relaxed) == 8);
+    }
+
+    {
+        puts("Testing <bit>.");
+        assert(popcount(0x1234ABCDu) == 15);
+        static_assert(popcount(0x1234ABCDu) == 15);
+    }
+
+    {
+        puts("Testing <bitset>.");
+        constexpr bitset<32> b{0x1234ABCDu};
+        assert(b[3] && b[2] && !b[1] && b[0]);
+        static_assert(b[3] && b[2] && !b[1] && b[0]);
+    }
+
+    {
+        puts("Testing <charconv>.");
+        char buf[4]{};
+        const to_chars_result result = to_chars(buf, end(buf), 3.14);
+        assert(result.ec == errc{});
+        assert(result.ptr == end(buf));
+#if 0 // TRANSITION, DevCom-1224512 (char_traits)
+        const string_view sv{buf, size(buf)};
+        assert(sv == "3.14");
+#else // ^^^ no workaround / workaround vvv
+        assert(buf[0] == '3' && buf[1] == '.' && buf[2] == '1' && buf[3] == '4');
+#endif // ^^^ workaround ^^^
+    }
+
+    {
+        puts("Testing <chrono>.");
+#ifdef MSVC_INTERNAL_TESTING // TRANSITION, DevCom-1159995 (UDLs)

```

Comment:
MSVC_INTERNAL_TESTING is a flag for stuff that only works with prod/fe right?

Reply:
That's correct. Only the Perl script used by `prod/fe` defines this macro - the Python script used by GitHub does not. This is specific to this test. See the PR description for more info.